boards/arduino-mega2560/include/periph_conf.h

```diff
@@ -264,23 +265,67 @@ extern "C" {
 /**
  * @brief GPIO configuration
  */
-#define GPIO_NUMOF          (0U)
-#define GPIO_0_EN           0
-#define GPIO_1_EN           0
-#define GPIO_2_EN           0
-#define GPIO_3_EN           0
-#define GPIO_4_EN           0
-#define GPIO_5_EN           0
-#define GPIO_6_EN           0
-#define GPIO_7_EN           0
-#define GPIO_8_EN           0
-#define GPIO_9_EN           0
-#define GPIO_10_EN          0
-#define GPIO_11_EN          0
-#define GPIO_12_EN          0
-#define GPIO_13_EN          0
-#define GPIO_14_EN          0
-#define GPIO_15_EN          0
+#define GPIO_NUMOF          12
+#define GPIO_2_EN           1
```

Comment:
The advantage of the periph_conf is that you can abstract the hardware-dependent stuff from your driver. That means the "GPIO_0" could for example either be port E pin 5 or port H pin 6... Please change this and all dependencies. You don't need to implement 15 GPIO pins (even if it would be nice :-) ). It is more important to prevent potential pin conflicts for later driver implementations. Tha means: Try to avoid implemention GPIOs when blocking other peripheral pins with that. Example: If there are just 3 pins that can be configured as SPI-pins, don't implement the GPIO functionality on them. Ok?